torchsparse/nn/modules/conv.py

```diff
@@ -6,25 +6,33 @@
 
 from ..functional import *
 
+from typing import Union
+
 __all__ = [
-    'Conv3d', 'ToBEVConvolution', 'ToBEVReduction', 'ToDenseBEVConvolution'
+    'Conv3d', 'ToBEVConvolution', 'ToBEVReduction', 'ToDenseBEVConvolution',
+    'ToBEVHeightCompression'
 ]
 
 
 class Conv3d(nn.Module):
     def __init__(self,
                  in_channels: int,
                  out_channels: int,
-                 kernel_size: int = 3,
-                 stride: int = 1,
+                 kernel_size: Union[int, list, tuple] = 3,
+                 stride: Union[int, list, tuple] = 1,
                  dilation: int = 1,
                  bias: bool = False,
                  transpose: bool = False) -> None:
         super().__init__()
         self.in_channels = inc = in_channels
         self.out_channels = outc = out_channels
         self.kernel_size = kernel_size
-        self.stride = stride
+        if isinstance(stride, int):
+            self.stride = stride
+        else:
+            assert isinstance(stride, list)
+            self.stride = torch.cuda.FloatTensor(stride).unsqueeze(0)
```

Comment:
The first point also applies to `kernel_size` and `dilation`.

Reply:
I fully agree with this idea. But I still think type checking and conversion might be necessary in `nn/functional/conv.py`, since users might directly use this interface and allowing them to passing different types of `stride` into the function can be a good idea.